### ISSUE TITLE
loss did not decrease

### ISSUE THREAD

**Issue by Mryangkaitong** [lvwerra/trl#193]
I applied trl to my own dataset, and print the loss at each step（https://github.com/lvwerra/trl/blob/main/trl/trainer/ppo_trainer.py#L682）, and found that the loss is difficult to decrease, even in the first few steps, it is useless to find a significant reduction, which means that the model does not seem to try Convergence, what is the general reason for this? Or can you give some advice? Or which indicator in the trl code can see whether the model is converging? Thank you so much

**Comment by younesbelkada** [comment]
Hello @Mryangkaitong 
Thanks for the issue! 
In fact this can depend on a lot of factors, firstly being the "domain gap" between your target dataset and the pre-trained model. 
What is the purpose of your dataset and what is the model that you are using? You might need to first pre-train your model on a dataset that is similar to yours. 
Also note that in the examples we have used a LR that is adapted to the model that has been pre-trained on imdb. Hence, you might need to play with that parameter a lot (i.e. use higher learning rate). You can also play with the KL penalty coefficient: https://github.com/lvwerra/trl/blob/a05ddbdd836d3217c80a4b3e679ba984bfd4fa24/trl/trainer/ppo_config.py#L84 to give the model more degree of freedom and deviate from its original distribution
Let us know if this helps!

**Comment by lvwerra** [comment]
Hi @Mryangkaitong, note that in RL the loss is less an indicator of model convergence than in normal supervised learning. It is more important to look at the distribution of rewards and how they shift throughout training.

**Comment by Mryangkaitong** [comment]
Thank you very much for your reply. I have fintune my model first (similar to a chatgpt model), and now I want to use the ppo algorithm to further optimize it. In order to test whether trl is effective, my reward model is very simple, just a rule: For example, when the reply has the special character <n> (this can be understood as a newline symbol, which means that the response is replied in sections, which is considered good). Specifically reward = response.count("<n>")*2

Another change is here:
https://github.com/lvwerra/trl/blob/main/trl/trainer/ppo_trainer.py#L610
Because my model logits and decoder_input_ids can correspond one-to-one, so my changes here are like this：

<img width="1335" alt="企业微信截图_319a1b05-d692-4f14-bd17-ff77044f7a3b" src="https://user-images.githubusercontent.com/23132307/223302928-6b3c2e9c-854f-460c-88ae-206039d00b91.png">

I have debugged init_kl_coef (increase or decrease) and lr and other parameters are not working, my configuration is as follows:

`
config = PPOConfig(
model_name="my_model",
learning_rate=5e-5,
batch_size=16,
ppo_epochs=1,
init_kl_coef=0.3,
log_with="wandb",
remove_unused_columns=False,
mini_batch_size=8
 )`

The current result is：

<img width="1183" alt="截屏2023-03-07 上午10 23 10" src="https://user-images.githubusercontent.com/23132307/223303178-38dbe152-f761-4d4d-9a97-6bb601333f35.png">

<img width="1528" alt="截屏2023-03-07 上午10 24 27的副本" src="https://user-images.githubusercontent.com/23132307/223303330-52388be6-3bce-4d94-9d30-3e97040ac8ad.png">

Now for all prompts, the responses are now exactly the same

**Comment by Mryangkaitong** [comment]
I solved it, thanks

**Comment by parshinsh** [comment]
@Mryangkaitong I'm facing the same issue. Can you please explain more how did you solve this?
